common/metrics/structured/base.go

```diff
@@ -0,0 +1,106 @@
+package structured
+
+import (
+	"maps"
+	"testing"
+	"time"
+
+	"github.com/uber-go/tally"
+	"go.uber.org/fx"
+)
+
+var Module = fx.Provide(func(s tally.Scope) Emitter {
+	return Emitter{scope: s}
+})
```

Comment:
this is the only constructor, outside tests.  makes it fairly likely that I've got my fx / resource / context / etc injection correct.